src/main/java/io/dropwizard/kafka/KafkaTopicsFactory.java

```diff
@@ -0,0 +1,61 @@
+package io.dropwizard.kafka;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+import io.dropwizard.jackson.Discoverable;
+import org.apache.kafka.clients.admin.NewTopic;
+import org.hibernate.validator.constraints.NotEmpty;
+
+import javax.validation.constraints.Min;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Map;
+import java.util.Objects;
+import java.util.stream.Collectors;
+
+public class KafkaTopicsFactory extends ArrayList<KafkaTopicsFactory.KafkaTopic> implements Discoverable {
```

Comment:
I'd say instead of inheritance, we should prefer composition.

The `KafkaTopicsFactory` can contain a field that is `List<KafkaTopic>`

Reply:
I agree extending looks bad, though I was aiming for the Configuration file not needing to look like this

```java
public class ExampleConfiguration {
    KafkaTopicsFactory topics;
}
```

```yaml
topics:
   topics: 
     - name: ...
```

Is there  a way around that nesting?

Reply:
Not the way it's currently structured.

Why not restructure it such that it looks more like this:

```java
public class ExampleConfiguration {
    @Valid
    @MinSize(1)
    @JsonProperty
    List<KafkaTopicsFactory> topics;
}
```

```yaml
topics:
  - name: ...
```

Reply:
I.e., instead of having the `KafkaTopicsFactory` contain the list, just have a _list of_ `KafkaTopicsFactory`s in your app's configuration.

And, if you go that route, maybe rename it `KafkaTopicFactory` (remove the plural for topic)